### PR TITLE
Fix response status if not found (#39)

### PR DESCRIPTION
**Pull request by yichen88**
Signed-off-by: yichen88 <tang.yichenyves@gmail.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
#39 

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Bug fix

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: afs-ws/afs-ws-client-utils/src/main/java/com/powsybl/afs/ws/client/utils/ClientUtils.java [hunk 1]
```diff
@@ -114,16 +113,11 @@ public static void checkOk(Response response) {
 
     public static <T> Optional<T> readOptionalEntityIfOk(Response response, Class<T> entityType) {
         Response.Status status = Response.Status.fromStatusCode(response.getStatus());
-        if (status == Response.Status.OK) {
-            return Optional.of(readEntityAndLog(response, entityType));
-        } else if (status == Response.Status.NO_CONTENT) {
+        if (status == Response.Status.NO_CONTENT || status == Response.Status.NOT_FOUND) {
             LOGGER.trace("    --> null");
             return Optional.empty();
```
Comment (jonenst) [inline afs-ws/afs-ws-client-utils/src/main/java/com/powsybl/afs/ws/client/utils/ClientUtils.java]: Why do you remove the INTERNAL_SERVER_ERROR case ?

File: afs-ws/afs-ws-client-utils/src/main/java/com/powsybl/afs/ws/client/utils/ClientUtils.java [hunk 2]
```diff
@@ -114,16 +113,11 @@ public static void checkOk(Response response) {
 
     public static <T> Optional<T> readOptionalEntityIfOk(Response response, Class<T> entityType) {
         Response.Status status = Response.Status.fromStatusCode(response.getStatus());
-        if (status == Response.Status.OK) {
-            return Optional.of(readEntityAndLog(response, entityType));
-        } else if (status == Response.Status.NO_CONTENT) {
+        if (status == Response.Status.NO_CONTENT || status == Response.Status.NOT_FOUND) {
```
Comment (jonenst) [inline afs-ws/afs-ws-client-utils/src/main/java/com/powsybl/afs/ws/client/utils/ClientUtils.java]: Maybe add a comment that the NO_CONTENT is for backwards compatibility with afs-server < version XX

File: afs-ws/afs-ws-client-utils/src/main/java/com/powsybl/afs/ws/client/utils/ClientUtils.java (hunk 1 above)
Comment (jonenst) [reply afs-ws/afs-ws-client-utils/src/main/java/com/powsybl/afs/ws/client/utils/ClientUtils.java]: and the UnexpectedResponse case ?

Comment (yichen88) [reply afs-ws/afs-ws-client-utils/src/main/java/com/powsybl/afs/ws/client/utils/ClientUtils.java]: these cases are still checked in `readEntityIfOk()`.

**Comment by mathbagu** [comment]
@yichen88 Could you please solve the DCO
@jonenst Could you complete the review please?

File: afs-ws/afs-ws-client-utils/src/main/java/com/powsybl/afs/ws/client/utils/ClientUtils.java (hunk 1 above)
Comment (jonenst) [reply afs-ws/afs-ws-client-utils/src/main/java/com/powsybl/afs/ws/client/utils/ClientUtils.java:120]: Ah yes sorry, didn't see it.

**Comment by jonenst** [comment]
Thanks, sorry for the delay !

**Pull request merged**